### PR TITLE
Second ESLint pass

### PR DESCRIPTION
**Pull request by adroitwhiz**
- Fix `npm run lint` to recurse into subdirectories
- Remove old no-longer-applicable tslint rules
- Disable problematic rules
- Make toScratchJS.ts and toScratchblocks.ts pass lint

### CODE REVIEW

File: src/io/scratch-js/toScratchJS.ts [hunk 1]
```diff
@@ -622,6 +624,8 @@ export default function toScratchJS(
               return `(new Date().getMinutes())`;
             case "SECOND":
               return `(new Date().getSeconds())`;
+            default:
+              return `('')`;
```
Comment (adroitwhiz) [inline src/io/scratch-js/toScratchJS.ts]: If this `default` doesn't exist, everything will fall through. An empty string matches 2.0 behavior for when this menu is set to an invalid value.

**Comment by adroitwhiz** [comment]
I can do that--one sec

**Comment by PullJosh** [comment]
> one sec

One... 😉

**Comment by adroitwhiz** [comment]
I think any further warning fixes will require more significant changes.

**Pull request merged**